Server/controllers/authController.js

```diff
@@ -33,6 +33,24 @@ const issueToken = (payload, appSettings) => {
   }
 };
 
+/**
+ * Creates and returns a JWT refresh token with an arbitrary payload
+ * @function
+ * @param {Object} payload - Here we have passed MongoDB assigned _id.
+ * @param {Object} appSettings
+ * @returns {String}
+ * @throws {Error}
+ */
+const issueRefreshToken = (payload, appSettings) => {
+  try {
+    const tokenTTL = appSettings.refreshTokenTTL ? appSettings.refreshTokenTTL : "7d";
+    return jwt.sign(payload, appSettings.refreshTokenSecret, { expiresIn: tokenTTL });
+  } catch (error) {
+    throw handleError(error, SERVICE_NAME, "issueRefreshToken");
+  }
+};
+
```

Comment:
As we discussed, this method is nearly identical to `issueToken`, they shoudl be refactored and combined into one method for issuing tokens.

Also a refresh token should not have a payload, it should consist only of an expiry date, no other information is necessary.

Reply:
Sure, I'll refactor the issueToken function.

If we don’t include the payload in the refresh token, how will I retrieve the payload when generating a new JWT access token? (We are including payload while generating access token.)

Reply:
> Sure, I'll refactor the issueToken function.
> 
> If we don’t include the payload in the refresh token, how will I retrieve the payload when generating a new JWT access token? (We are including payload while generating access token.)

The expired access token will be sent along with the refresh token, so you can grab the payload from the access token

Reply:
Oh got it! 

@ajhollid thanks for clarity.